bitnami/mongodb-sharded/templates/mongos/mongos-deployment.yaml

```diff
@@ -0,0 +1,290 @@
+                - "db.adminCommand('ping')"
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          {{- end }}
+          {{- if .Values.readinessProbe.enabled }}
+          readinessProbe:
+            exec:
+              command:
+                - mongo
+                - --eval
+                - "db.adminCommand('ping')"
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+          {{- end }}
+          volumeMounts:
+            {{- if .Values.usePasswordFile }}
+            - name: secrets
+              mountPath: /bitnami/mongodb/secrets/
+            {{- end }}
+            {{- if .Values.mongos.initScriptsCM }}
+            - name: custom-init-scripts
+              mountPath: /docker-entrypoint-initdb.d/configmap
+            {{- end }}
+            {{- if .Values.mongos.initScriptsSecret }}
+            - name: custom-init-scripts-secrets
+              mountPath: /docker-entrypoint-initdb.d/secret
+            {{- end }}
+            {{- if or .Values.mongos.config .Values.mongos.configCM }}
+            - name: config
+              mountPath: /bitnami/mongodb/conf/
+            {{- end }}
+            {{- if $.Values.common.initScriptsCM }}
+            - name: custom-init-scripts-cm
+              mountPath: /docker-entrypoint-initdb.d/cm
+            {{- end }}
+            {{- if $.Values.common.initScriptsSecret }}
+            - name: custom-init-scripts-secret
+              mountPath: /docker-entrypoint-initdb.d/secret
+            {{- end }}
+            {{- if $.Values.common.extraVolumeMounts }}
+              {{- include "mongodb-sharded.tplValue" ( dict "value" $.Values.common.extraVolumeMounts "context" $ ) | nindent 12 }}
+            {{- end }}
+            {{- if $.Values.mongos.extraVolumeMounts }}
+              {{- include "mongodb-sharded.tplValue" ( dict "value" $.Values.mongos.extraVolumeMounts "context" $ ) | nindent 12 }}
+            {{- end }}
+          resources: {{- toYaml .Values.mongos.resources | nindent 12 }}
+        {{- if .Values.metrics.enabled }}
+        - name: metrics
+          image: {{ include "mongodb-sharded.metrics.image" . }}
+          imagePullPolicy: {{ .Values.metrics.image.pullPolicy | quote }}
+          {{- if .Values.securityContext.enabled }}
+          securityContext:
+            runAsNonRoot: true
```

Comment:
If we ever want to run this as root we would need to edit the template. Can you parametrize this?